raft/log.go

```diff
@@ -211,20 +216,26 @@ func (l *raftLog) term(i uint64) uint64 {
 		return t
 	}
 	if err == ErrCompacted {
-		return 0
+		return math.MaxUint64
```

Comment:
MaxUint64 is also a valid term, it's just one that's unlikely to ever be reached. Unsigned ints don't give us any invalid values to work with so I think it would be best to add an error return to this method.